src/movies.js

```diff
@@ -1,16 +1,65 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
-
+function getAllDirectors(directores){
+const directorsMovies = directores.map(function(oneMovie){
+    return oneMovie.director;
+})
+return directorsMovies;
+}
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-
+// no reusar Get all directors de momento
+function howManyMovies(movies){
+    var spielbergmovies = movies.filter(function (OneElement){
+        return OneElement.director === "Steven Spielberg" && OneElement.genre.includes("Drama")
+         })
+    return spielbergmovies.length
+}
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+// Capu dice que con reduce deberia ser suficiente
+
+function ratesAverage(movies){
+    var SumOfRates =  movies.reduce((sum, movi) => sum + movi.rate, 0);
+    var mediaRate = SumOfRates/movies.length;
+    var redondear = Math.ceil(mediaRate* 100) / 100;
+    if (SumOfRates === 0) return 0 ;
+    return redondear;
+ }
```

Comment:
bien! Y para el test "Return average even if one of the movies does not have rate!", puedes añadir un condicional en la función callback del método reduce(), en caso de que movi.rate no exista:

var SumOfRates =  movies.reduce((sum, movi) => {
        if(!movi.rate) return sum
        else return sum + movi.rate
      }, 0);